pyasic/rpc/btminer.py

```diff
@@ -559,11 +560,24 @@ async def set_normal_power(self) -> dict:
         """
         return await self.send_privileged_command("set_normal_power")
 
-    async def update_firmware(self):  # noqa - static
-        """Not implemented."""
-        # to be determined if this will be added later
-        # requires a file stream in bytes
-        return NotImplementedError
+    async def upgrade_firmware(self, firmware: bytes):
```

Comment:
This should be called `update_firmware`, to be consistent with the API docs from whatsminer.  The miner RPC classes should be a direct analog to the RPC handler on the miner as much as possible.

The `upgrade_firmware` method needs to be implemented on a miner in `pyasic.miners.backends`, in this case, `pyasic.miners.backends.btminer.BTMiner`.

Reply:
I have renamed the method to `update_firmware` to align with the documentation from Whatsminer, 
I also implemented the `upgrade_firmware` method within the `pyasic.miners.backends.btminer.BTMiner` class.